client/chat/channel-banner-reducer.ts

```diff
@@ -0,0 +1,56 @@
+import { Immutable } from 'immer'
+import { ChannelBannerId, ChannelBannerJson } from '../../common/chat-channels/channel-banners'
+import { immerKeyedReducer } from '../reducers/keyed-reducer'
+
+export interface ChannelBannerState {
+  /** An array of channel banners that are available to us. */
+  availableChannelBanners: ChannelBannerId[]
```

Comment:
I think this stuff is likely to change now anyway, but wasn't the whole idea that banners would be available to a *channel* not a *user*? Thus you would need to key this list per channel